cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -229,6 +229,17 @@ class ORCFileReader::Impl {
     return GetArrowSchema(type, out);
   }
 
+  Result<std::shared_ptr<KeyValueMetadata>> ReadMetadata() {
+    std::list<std::string> keys = reader_->getMetadataKeys();
+    auto metadata = std::make_shared<KeyValueMetadata>();
+    if (!keys.empty()) {
```

Comment:
Why not `for (const auto& key : keys) {`?

Reply:
Sure!